### DISCUSSION TITLE
How to fetch table from a localhost postgresql database as a datasource in redash?

### DISCUSSION THREAD

**Discussion by RYees** [getredash/redash#6692] [General]
After a lot of trials i managed to connect my localhost postgresql to my redash as a data source code, but two issues happen 
1. Redash is only able to connect to my localhost default database that is postgres, it refuse to connect to other databases i created, it throws 'database_name' does not exist everytime. but it connect with default one

2. Even after connection with the default one, i create a table with values inside the public schema and try to see if the table will be fetched automatically to start query, but redash return an empty schema, or doesn' load the tables found in the default database even after a successful connection with the default database

I have checked everything on my database, both on pgadmin and psql command the tables does exist on the default postgres database and the databases does exist in my localhost postgresql

what am i missing? what could be the reason for these?

**Comment by justinclift** [comment]
It *sounds* like you're connecting to the Redash PostgreSQL database (somehow), rather than the one you're wanting to.

At least, that would explain why the things you're expecting in your PostgreSQL database don't seem to be present when you look via Redash. :frowning:

**Reply by justinclift** [reply]
For the "how to get this working?" part of things, you'll need to figure out some way to connect Redash running in it's containers to your local computer.

That will probably need some discussion, so we can figure out what could work. :smile:

Is your Redash installation on other computer, or is it running on your desktop as well?

**Comment by RYees** [comment]
Thank you for responding
so what is did is i have made the Redash installation on ubuntu os that is installed on a virtual box, the localhost postgresql is also installed in the virtual machine and the way redash able to find the localhost postgresql and connect with the default database is by using the default ip address of the ubuntu on the virtual machine as the host address,

host:  the ip address
port: 5432
user:  postgres
password:  databasepassword
database: postgres

these the configuration i used to connect the redash with the localhost postgres in my virtual box desktop machine

**Reply by justinclift** [reply]
Hmmm, that's a bit confusing still. :wink:

Could you be ok to run this command on your Ubuntu VM, and paste the result here?

```
grep -A4 listen_addresses /etc/postgresql/16/main/postgresql.conf
```

That *should* (if I guessed the path correctly) tell us what network interface PostgreSQL is listening on, which will help me understand things. :smile:

**Comment by RYees** [comment]
Sure here is it:
vboxuser@ubuntu:~$ grep -A4 listen_addresses /etc/postgresql/14/main/postgresql.conf
listen_addresses = '*'		# what IP address(es) to listen on;
					# comma-separated list of addresses;
					# defaults to 'localhost'; use '*' for all
					# (change requires restart)
port = 5432				# (change requires restart)


but i noticed that it is not connecting with my actual localhost postgresql database, i think it is still connecting with redash postgres database that is running under port 5432 which is specified under the docker-compose.yml file I still don't know why it is accepting my localhost postgres database password, anyway the way i come to these conclusion is because when i change my localhost postgresql port to 5434, it doesn't find it with the new port number and throws the below error

Connection Test Failed:
Connection refused Is the server running on that host and accepting TCP/IP connections? 

while my localhost is at port 5434, when i test the postgresql datasource on redash under port 5432, it pass with success

**Reply by justinclift** [reply]
No worries.  With your `postgresql.conf` file, what happens if you change this line:

```
listen_addresses = ''
```

... to this:

```
listen_addresses = '*'
```

... then restart the database?

I'm expecting that will tell your PostgreSQL database (not the internal Redash one) to listen on every available network interface it can.

That *might* make it available on an IP address that Redash can access, as `localhost` can be difficult to access across VM's and docker containers.

That's my first thought anyway. :smile:

**Comment by RYees** [comment]
I did that but still the same.  Maybe I should use heroku's free postgresql database, I am wasting too much time with these one.

**Reply by justinclift** [reply]
Yeah, it's probably the better approach.  "Pick your battles" kind of thing. :smile:

**Comment by RYees** [comment]
Thank you for accompanying me through these issues. GREATLY APPRECIATED😄👍🙏

**Reply by justinclift** [reply]
No worries at all. :smile:
